docs/en/sql-reference/functions/null-functions.md

```diff
@@ -96,6 +137,36 @@ Result:
 └───┘
 ```
 
+## isNotDistinctFrom
+
+Performs null-safe comparison. Used to compare JOIN keys which contain NULL values in the JOIN ON section.
+This function will consider two `NULL` values as identical and will return `true`, which is distinct from the usual
+equals behavior where comparing two `NULL` values would return `NULL`.
+
+:::note
+This function can only be used as part of a JOIN ON section of a query. The function is unimplemented for any other use.
```

Comment:
To the reader it will not be clear what "is unimplemented for any other use case" means and implies.

Perhaps we should say: `This function is an internal function used by the implementation of JOIN ON. Please do not use it manually in queries."